zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -384,19 +384,15 @@ public void addUriDomainMapNode() {
         if (!X509AuthenticationConfig.getInstance().isX509ZnodeGroupAclEnabled()) {
             return;
         }
-        DataNode zookeeperZnode = nodes.get(procZookeeper);
-        if (zookeeperZnode != null) { // should always be the case
+        try {
+            DataNode zookeeperZnode = nodes.get(procZookeeper);
             zookeeperZnode.addChild(uriDomainMapChildZookeeper);
-        } else {
-            assert false : "There's no /zookeeper znode - this should never happen.";
-        }
 
-        nodes.put(uriDomainMapZookeeper, new DataNode(new byte[0], -1L, new StatPersisted()));
-        try {
-            // Reconfig node is access controlled by default (ZOOKEEPER-2014).
+            nodes.put(uriDomainMapZookeeper, new DataNode(new byte[0], -1L, new StatPersisted()));
             setACL(uriDomainMapZookeeper, ZooDefs.Ids.OPEN_ACL_UNSAFE, -1);
-        } catch (KeeperException.NoNodeException e) {
-            assert false : "There's no " + uriDomainMapZookeeper + " znode - this should never happen.";
+            LOG.info("Successfully created client URI-domain root path :{}", uriDomainMapZookeeper);
+        } catch (Exception e) {
```

Comment:
Should we fail here if the root path creation is failed?

Reply:
If we fail here itself then customer won't even be able to use any other authentication provider or plaintext port. Hence i just added log here but raising an exception in ZkClientUriDomainMappingHelper constructor if root path is not present. Does it make sense? @mgao0

Reply:
Make sense. Thank you!